### PR TITLE
fix: 5003 nft card without name

### LINKED ISSUE #5003: NFT card without name
image
- you can see it here https://kodadot.xyz/rmrk/gallery/16619627-EC306760418F319267-JA_TPPTZ-DWL2--0000000000000002

- should have same dimensions as the normal ones.

### PR DESCRIPTION
**Pull request by maderi21**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context

- [x] Closes #5003 
- [ ] Requires deployment <>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [x] I've tested it at </bsx/collection>
- [x] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=JHJGu2aWyAFBfcso8bR7ma82p15YAavyN6vQfJyWuhniNGa)

#### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.
image

### CODE REVIEW

**Comment by maderi21** [comment]
@preschian still sometimes getting this btw:
image

**Comment by maderi21** [comment]
> add more fallback maybe?

added

**Comment by maderi21** [comment]
possibly a problem with layout in explore (this would be an issue even if user selected very long collection name). maybe separate issue? 
image

File: components/rmrk/Gallery/GalleryCard.vue [hunk 1]
```diff
@@ -83,6 +83,7 @@ export default class GalleryCard extends mixins(AuthMixin) {
   async fetch() {
     if (this.metadata) {
       const meta = await processSingleMetadata<NFTMetadata>(this.metadata)
+      console.log(meta)
```
Comment (Jarsen136) [inline components/rmrk/Gallery/GalleryCard.vue:86]:
```suggestion
```

File: components/rmrk/Gallery/GalleryCard.vue [hunk 2]
```diff
@@ -34,7 +34,7 @@
           :class="{ 'title is-4': largeDisplay }"
           :title="name">
           <div class="has-text-overflow-ellipsis">
-            {{ nftName }}
+            {{ nftName || id }}
```
Comment (Jarsen136) [inline components/rmrk/Gallery/GalleryCard.vue:37]:
```
get nftName(): string {
    return this.name || this.title || this.id
  }
```

**Review by Jarsen136 — COMMENTED** [review]
otherwise code lgtm

File: components/rmrk/Gallery/GalleryCard.vue (hunk 2 above)
Comment (maderi21) [reply components/rmrk/Gallery/GalleryCard.vue]: done

**Comment by exezbcz** [comment]
what about putting -- instead of the numbers?

**Comment by maderi21** [comment]
> what about putting -- instead of the numbers?

as you wish, I've used '--' instead of the IDs.

**Comment by exezbcz** [comment]
> > what about putting -- instead of the numbers?
> 
> as you wish, I've used '--' instead of the IDs.

and in the gallery item it should apply as well or not yet?

**Comment by maderi21** [comment]
> > > what about putting -- instead of the numbers?
> > 
> > 
> > as you wish, I've used '--' instead of the IDs.
> 
> and in the gallery item it should apply as well or not yet?

I don't really know, but I guess your question was a request :D ? 

image

Please let me know if anything else is required here, I'm not sure if empty dashes work better than some ID.

**Comment by maderi21** [comment]
> soo, lets leave the id there and then fix the option to mint nft without name :D

I think I have fixed it elsewhere, although there may be some other problem I've seen conversation about. People won't be allowed to pass empty strings anymore, that should be clear now. Anyways, this PR is, I guess, done now.

**Review by vikiival — APPROVED** [review]

**Comment by vikiival** [comment]
pay 20 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $20 @ 44.89 USD/KSM ~ 0.446 $KSM
🧗 [JHJGu2aWyAFBfcso8bR7ma82p15YAavyN6vQfJyWuhniNGa](https://kusama.subscan.io/account/JHJGu2aWyAFBfcso8bR7ma82p15YAavyN6vQfJyWuhniNGa)
🔗 [0x13f31e15340e615131a5059474b062920c57f1bc6ee1136d471ec6d7c71952b6](https://kusama.subscan.io/extrinsic/0x13f31e15340e615131a5059474b062920c57f1bc6ee1136d471ec6d7c71952b6)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**